test/unit/client.js

```diff
@@ -62,3 +62,48 @@ describe('Client userMuteStatus', function () {
 		expect(client.userMuteStatus('missingUser')).not.to.be.ok;
 	});
 });
+
+describe('Detect node environment', () => {
+	const client = new StreamChat('', '');
+	it('node property should be true', () => {
+		expect(client.node).to.be.true;
+	});
+
+	it('should warn when using connectUser on a node environment', async () => {
+		const _warn = console.warn;
+		let warning = '';
+		console.warn = (msg) => {
+			warning = msg;
+		};
+
+		try {
+			await client.connectUser({ id: 'user' }, 'fake token');
+		} catch (e) {}
+
+		await client.disconnect();
+		expect(warning).to.equal(
+			'Please do not use connectUser server side. If you have a valid use-case, add "allowServerSideConnect: true" to the client options to disable this warning.',
```

Comment:
Perhaps we should mention the billing implications here?

'Please do not use connectUser server side. connectUser impacts MAU and concurrent connection usage and thus your bill. If you have a valid use-case, add "allowServerSideConnect: true" to the client options to disable this warning.'